public/app/features/datasources/settings/DataSourceSettingsPage.tsx

```diff
@@ -209,7 +210,7 @@ export class DataSourceSettingsPage extends PureComponent<Props, State> {
           <div>
             <div className="gf-form-button-row">
               {showDelete && (
-                <button type="submit" className="btn btn-danger" onClick={this.onDelete}>
+                <button type="submit" className="btn btn-danger" onClick={event => this.onDelete(event)}>
```

Comment:
Why is the delete button a submit button, can’t it be a normal button and this problem would go away?

Reply:
Yes, it can be a normal button, but I  think it is better to be consistent with the save button.

Reply:
Yes, of course. But before, I want it to be consistent with the save button.
Now,I have resolved .